galite-core/src/main/kotlin/org/kopi/galite/visual/VlibProperties.kt

```diff
@@ -18,8 +18,70 @@
 
 package org.kopi.galite.visual
 
+
+import java.util.Locale
+import java.text.MessageFormat
+import org.kopi.galite.util.base.InconsistencyException
+import org.kopi.galite.l10n.LocalizationManager
+
+/**
+ * This class handles localized properties
+ */
 object VlibProperties {
-  fun getString(key: String): String {
-    TODO()
+
+  // ----------------------------------------------------------------------
+  // STATIC METHODS
+  // ----------------------------------------------------------------------
+
+  /**
+   * Returns the value of the property for the given key.
+   *
+   * @param     key             the property key
+   * @return    the requested property value
+   */
+  fun getString(key: String): String = getString(key, null)
+
+  /**
+   * Returns the value of the property for the given key.
+   *
+   * @param     key             the property key
+   * @param     param1          the first property parameter
+   * @param     param1          the second property parameter
+   * @return    the requested property value
+   */
+  fun getString(key: String, param1: Any, param2: Any): String {
+    return getString(key, arrayOf(param1, param2))
   }
+
+  fun getString(key: String, params: Any?): String {
+    val format: String
+    val manager: LocalizationManager = if (ApplicationContext.applicationContext!!.getApplication() != null) {
+      ApplicationContext.getLocalizationManager()
+    } else {
+      LocalizationManager(Locale.getDefault(), null)
+    }
+    return try {
+      // Within a String, "''" represents a single quote in java.text.MessageFormat.
+      format = manager.getPropertyLocalizer(VLIB_PROPERTIES_RESOURCE_FILE, key).getValue().replace("'", "''")
+
+      if (params is Array<*>) {
+        MessageFormat.format(format, *params)
+      } else {
+        MessageFormat.format(format, params)
+      }
+
+    } catch (e: InconsistencyException) {
+      ApplicationContext.reportTrouble("localize Property",
+              "org.kopi.galite.visual.VlibProperties.getString(String key, Object[] params)",
+              e.message!!,
+              e)
+      System.err.println("ERROR: " + e.message)
+      "!$key!"
+    }
+  }
+
+  // ----------------------------------------------------------------------
+  // DATA MEMBERS
+  // ----------------------------------------------------------------------
+  private const val VLIB_PROPERTIES_RESOURCE_FILE = "org/kopi/galite/lib/resource/VlibProperties"
```

Comment:
Could you fix this to "resource/org/kopi/galite/VlibProperties"

Reply:
done